### DISCUSSION TITLE
Saving data from mulitple timesteps in user-scripts

### DISCUSSION THREAD

**Discussion by dylanjaybaxter** [foxglove/community#1043] [Q&A]
Hello,

I am looking to perform an operation on data in a prerecorded mcap file where I compute a result based on both the current and previous timestep. Is there a way to do this in user scripts? From my experimentation so far it looks like variables in the global scope are being reinitialized each time a topic is read. 

Please let me know if this is possible.

Thanks,
Dylan

**Comment by jtbandes** [comment]
Yes, I believe you should be able to save a variable and read the old value from the previous invocation of the script. Could you share more what you have tried/seen when you said this:

>  From my experimentation so far it looks like variables in the global scope are being reinitialized each time a topic is read.
